jwst/outlier_detection/outlier_detection_step.py

```diff
@@ -69,13 +71,16 @@ class OutlierDetectionStep(Step):
         scale_detection = boolean(default=False)
         search_output_file = boolean(default=False)
         allowed_memory = float(default=None)  # Fraction of memory to use for the combined image.
-        in_memory = boolean(default=False)
```

Comment:
I'm not sure I like the idea of removing the `in_memory` param from the step. What if, for example, a user wants to avoid memory use (`in_memory = False`), but also doesn't want or need any of the intermediate files to persist after the step is done (`save_intermediate_results = False`). With this change they have no way to do that. If they don't want the intermediate files to persist, they are forced to have the step run with all the data in memory. If they want to conserve memory (or there isn't enough memory to run it that way), it's OK to have the intermediate files appear temporarily, you just want to clean them up at the end.

I would rather see both params remain in use and have the new code (to delete the files at the end) get activated when `in_memory = False` and `save_intermediate_files = False`.

Reply:
done

Reply:
I removed the if in memory requirement for deleting the files. This is only controlled by save_intermediate_results, though the user can choose to set in_memory = Files, and this will generate the files